app/models/exercise/playground.rb

```diff
@@ -8,7 +8,7 @@ def setup_query_assignment!(assignment)
   end
 
   def save_query_results!(assignment)
-    unless assignment.passed?
+    unless assignment.solved?
```

Comment:
I would keep this one as just `passed?` so if a playground was skipped the student can still retry it and make it passed!.
Although we haven't thought about the interface yet, we should assume that we will let them do that.

Reply:
Sounds right, I'm reverting this change.